install_deps_ubuntu_22.04.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -e
+
+apt-get update
+apt-get install -y curl gnupg
+echo 'deb http://download.opensuse.org/repositories/home:/p4lang/xUbuntu_22.04/ /' | tee /etc/apt/sources.list.d/home:p4lang.list
+curl -fsSL https://download.opensuse.org/repositories/home:p4lang/xUbuntu_22.04/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/home_p4lang.gpg > /dev/null
+
+apt-get update
+
+apt-get install -qq --no-install-recommends \
+    automake \
+    build-essential \
+    cmake \
+    git \
+    g++ \
+    libboost-dev \
+    libboost-filesystem-dev \
+    libboost-program-options-dev \
+    libboost-system-dev \
+    libboost-thread-dev \
+    libgmp-dev \
+    libgrpc++-dev \
+    libgrpc-dev \
+    libopenmpi-dev \
+    libnanomsg-dev \
+    libpcap-dev \
+    libprotobuf-dev \
+    libprotoc-dev \
+    libssl-dev \
+    libthrift-0.16.0 \
+    libthrift-dev \
+    libtool \
+    pkg-config \
+    protobuf-compiler \
+    protobuf-compiler-grpc \
+    python3-all \
+    python3-six \
+    python3-thrift \
+    thrift-compiler \
+    libreadline-dev \
+    p4lang-pi
```

Comment:
could you remind me if this package is automatically updated when the main branch of the p4lang/PI repo is updated?
sometimes, some changes to bmv2 require the latest p4lang/PI code

Reply:
I test and update this package manually as it could occasionally break due to changes in the PI dependencies.
This is to ensure that when users run `apt-get upgrade` the updated version would work as expected.

The build scripts the packages are available at https://github.com/rst0git/p4lang-packages. I'm using these packages in my development environment and intend to continue maintain them for the foreseeable future.

I am happy to apply any changes from the main branch of the p4lang/PI repository and update the package as necessary.

Reply:
We are also using these packages in the CI environment for p4c [^1] and to build the tutorials VM [^2].

[^1]: https://github.com/p4lang/p4c/blob/daa55c88/tools/ci-build.sh#L109
[^2]: https://github.com/p4lang/tutorials/blob/master/vm-ubuntu-20.04/root-release-bootstrap.sh#L43

Reply:
it's a bit different, as p4lang/behavioral-model and p4lang/PI are strongly correlated (PI defines some C functions which are implemented in bmv2). But we can see how it goes. It's not like breaking updates are very frequent these days.